File: LeetCode_Monthly_Challenge/June/Python3/Week_1/invert_binary_tree.py
"""
PROBLEM:
Invert a binary tree.
Example:
Input:

     4
   /   \
  2     7
 / \   / \
1   3 6   9
Output:
     4
   /   \
  7     2
 / \   / \
9   6 3   1

"""


# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right


class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        if root is None:
            return None
        l=self.invertTree(root.left)
        r=self.invertTree(root.right)
            
        root.right=l
        root.left=r
        return root

File: LeetCode_Monthly_Challenge/June/Java/Week_1/invert_binary_tree.java
/*
PROBLEM:
Invert a binary tree.
Example:
Input:

     4
   /   \
  2     7
 / \   / \
1   3 6   9
Output:
     4
   /   \
  7     2
 / \   / \
9   6 3   1
*/


/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */


class Solution {
    public TreeNode invertTree(TreeNode root) {
        if(root==null)
            return null;
        else{
            TreeNode temp;
            invertTree(root.left);
            invertTree(root.right);
            
            temp=root.left;
            root.left=root.right;
            root.right=temp;
        }
        return root;
    }
}